### DISCUSSION TITLE
Errors in lammps.py and pca.py docstrings

### DISCUSSION THREAD

**Discussion by nataliyah123** [MDAnalysis/mdanalysis#4454] [Q&A]
1. Where can I find lammps.data file?
2.  I fixed pca.py setup errors but now two more errors popped up 

```
Document: documentation_pages/analysis/pca
------------------------------------------
**********************************************************************
File "../../../MDAnalysis/analysis/pca.py", line ?, in default
Failed example:
    first_interval.rmsip(second_interval, n_components=3)
Expected:
    0.38147609331128324
Got:
    0.38147609331128335
**********************************************************************
File "../../../MDAnalysis/analysis/pca.py", line ?, in default
Failed example:
    pca.rmsip(first_interval.results.p_components.T,
              second_interval.results.p_components.T,
              n_components=3)
Expected:
    0.38147609331128324
Got:
    0.38147609331128335
**********************************************************************
1 items had failures:
   2 of  12 in default
12 tests in 1 items.
10 passed and 2 failed.
***Test Failed*** 2 failures.

```
values are almost the same. what should I do to get rid of this

**Comment by RMeli** [comment]
Hello.

1. Could you please give us more details? Which file are you referring to? What are you trying to do?
2. Given the very small difference, looks like numerical errors (probably the algorithm implementation changed in the backend, or similar). The "easy" way to deal with this would be to update the number, but it will certainly break again in the future. A better approach would be to follow what the [doctest documentation - warnings](https://docs.python.org/3/library/doctest.html#warnings) suggests. Save the result in a temporary variable, and print it with sensible rounding.

From [doctest documentation - warnings](https://docs.python.org/3/library/doctest.html#warnings):
```
>>> 1./7  # risky
0.14285714285714285
>>> print(1./7) # safer
0.142857142857
>>> print(round(1./7, 6)) # much safer
0.142857
```

**Comment by RMeli** [comment]
PS: please give discussions a sensible title, so that one knows what the discussion is about from the title. `lammps.py and pca.py` doesn't mean much; from just reading the title I have no idea what's going on.

**Reply by nataliyah123** [reply]
Is the title ok now,  lammps.py is in ` ..\..\package\MDAnalysis\coordinates` folder and pca.py is in ` ..\..\package\MDAnalysis\analysis`. I cannot find the lammps.data that was mentioned on #L69 and onwards in lammps.py. I am trying to run make doctest

**Comment by orbeckst** [comment]
Is this related to PR #4457 ?

If yes, then we can continue any discussions on the PR and close this post.

**Reply by nataliyah123** [reply]
@orbeckst PR [#4457](https://github.com/MDAnalysis/mdanalysis/pull/4457) has only the changes made in PCA.py. I have different branches for different modules (PCA, LAMMPS). But timestep, pca, lammps all are related to the mega issue [#3925](https://github.com/MDAnalysis/mdanalysis/issues/3925). I will make a small change in the lammps.py file and make a PR with [WIP] in the title and close this discussion.  Thank to @RMeli the PCA did pass the test.
